include/picongpu/plugins/adios/ADIOSWriter.hpp

```diff
@@ -1446,10 +1446,10 @@ class ADIOSWriter : public IIOBackend
 
         /* allocate buffer in MB according to our current group size */
         /* `1 + mem` minimum 1 MiB that we can write attributes on empty GPUs */
-        size_t writeBuffer_in_MiB=1+threadParams->adiosGroupSize / 1024 / 1024;
+        // size_t writeBuffer_in_MiB=1+threadParams->adiosGroupSize / 1024 / 1024;
         /* value `1.1` is the secure factor if we miss to count some small buffers*/
-        size_t buffer_mem=static_cast<size_t>(1.1 * static_cast<float_64>(writeBuffer_in_MiB));
-        adios_set_max_buffer_size(buffer_mem);
+        // size_t buffer_mem=static_cast<size_t>(1.1 * static_cast<float_64>(writeBuffer_in_MiB));
+        // adios_set_max_buffer_size(buffer_mem);
         threadParams->adiosBufferInitialized = true;
```

Comment:
Two potentially silly questions:
* why is the now obsolete code commented out, and not deleted?
* is `threadParams->adiosBufferInitialized` actually used anywhere (could not find the usage for this member at all)?

Reply:
The whole implementation that goes through the data first, stores IDs, estimates sizes, etc. is unfortunately outdated. That was what we needed to do around 2015.

We are planning to unify the ADIOS and HDF5 backend with a student in the next months via the new https://github.com/openPMD/openPMD-api 

But until then, I just want to apply fixes to it and don't invest time into the legacy part of those plugins.